staging/src/k8s.io/client-go/tools/cache/listers.go

```diff
@@ -32,12 +32,18 @@ type AppendFunc func(interface{})
 
 func ListAll(store Store, selector labels.Selector, appendFn AppendFunc) error {
 	for _, m := range store.List() {
-		metadata, err := meta.Accessor(m)
-		if err != nil {
-			return err
-		}
-		if selector.Matches(labels.Set(metadata.GetLabels())) {
+		if selector.Empty() {
```

Comment:
(applies to all) slight preference for computing `selectAll := selector.Empty()` outside the loop, and continuing early, like this:

```
if selectAll {
  appendFn(m)
  continue
}
...
```

Reply:
Fine with me. Applied.